oidc-signature-extension.md

```diff
@@ -262,200 +208,224 @@ The following is a non-normative example of the claims in a Request Object befor
 
-A request for signature MUST contain the [Signature Request Parameter Claim](#the-signature-request parameter-claim) and its inclusion in the request MUST follow the requirements stated in sections [3.2.1](#placement-of-the-parameter-in-an-authentication-request), "[Placement of the Parameter in an Authentication Request](#placement-of-the-parameter-in-an-authentication-request)" and [3.2.2](#security-requirements), "[Security Requirements](#security-requirements)".
+The authentication request MUST contain the `prompt` parameter<sup>1</sup> and its value MUST be include both the 
+`login` and `consent` parameter values.  The reason for this is that a signature must never be generated based on a previous authentication (`login`) and that the Relying Party wants to ensure that the user actually sees the sign message and understands
+that he or she is performing a signature operation (`consent`).
 
-The authentication request MUST contain the `prompt` parameter<sup>1</sup> and its value MUST be include the `login` parameter value.  The reason for this is that a signature must never be generated based on a previous authentication.
+The Relying Party SHOULD examine the discovery document regarding supported MIME types for the `sign_message` field of the
+`https://id.oidc.se/param/signRequest` request parameter value (see [section 5.3](#discovery)), and only use a MIME
+type supported by the OpenID Provider. If no such information is available in the OP discovery document, the Relying 
+Party SHOULD use the MIME type `text/plain` for the sign message. 
 
-> \[1\]: The `prompt` parameter can be provided either as an ordinary request parameter or as a field in a Request Object.
+> **\[1\]:** The `prompt` parameter can be provided either as an ordinary request parameter or as a field in a Request Object.
 
 <a name="requirements-on-signing-user"></a>
 #### 4.1.1. Requirements on Signing User
 
-In most cases a user is already logged into the service that wants the user to sign some date, for example an approval or a document. The data that is to be signed can be sensitive, and the service will need to ensure that only the intended user can view this data.  
+In most cases a user is already logged into the service that wants the user to sign some data, for example an approval or 
+a document. The data that is to be signed can be sensitive, and the service will need to ensure that only the intended user
+can view this data.
 
-TODO: Describe how to use the claims parameter to include information about the user that is to sign the data, for example, give the personal identity number. See chapter 5.5.1 of \[[OpenID.Core](#openid-core)\].
+If a Relying Party wanting to bind a signature operation to a particular identity SHOULD assign the necessary identity
```

Comment:
Nitpick: The wording should be either:

"A Relying Party wanting to bind a signature operation to a particular identity SHOULD assign ..."

or 

"If a Relying Party wanting to bind a signature operation to a particular identity, the Relying Party SHOULD assign ..."